frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { match as RouteMatch } from 'react-router';
+import { useTranslation } from 'react-i18next';
+import { Wizard, WizardStep } from '@patternfly/react-core';
+import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
+import { ListKind } from '@console/internal/module/k8s';
+import { CreateStorageSystemFooter } from './create-storage-system-footer';
+import { CreateStorageSystemHeader } from './create-storage-system-header';
+import { BackingStorage, createSteps } from './create-storage-system-steps';
+import { initialState, reducer, WizardReducer } from './reducer';
+import { StepsId, StorageClusterIdentifier } from '../../constants/create-storage-system';
+import { StorageSystemKind } from '../../types';
+import { StorageSystemModel } from '../../models';
+
+const CreateStorageSystem: React.FC<CreateStorageSystemProps> = ({ match }) => {
+  const { t } = useTranslation();
+  const [state, dispatch] = React.useReducer<WizardReducer>(reducer, initialState);
+  const [ssList] = useK8sGet<ListKind<StorageSystemKind>>(StorageSystemModel);
+
+  const {
+    params: { appName },
+    url,
+  } = match;
+
+  const existingStorageSystems: Set<StorageSystemKind['spec']['kind']> =
```

Comment:
```suggestion
  const existingStorageSystems: Set<string> =
```
We don't need to specify it as `StorageSystemKind['spec']['kind']` since we know it is a string.
We are also performing string operations in below code. if kind type changes in the future, it will have compilation error here which will be useful for fixing.

also makes it more readable

Reply:
Exactly :), id kind of type changes, it should fail.
We should always define types in one place and use that everywhere. I see readability is not much affected.

Reply:
I have redefined it for readability:
```
const formatStorageSystemList = (storageSystems: StorageSystemKind[] = []): StorageSystemSet =>
  storageSystems.reduce(
    (kinds: StorageSystemSet, ss: StorageSystemKind) => kinds.add(ss.spec.kind),
    new Set(),
  );
  ```